include/wil/registry.h

```diff
@@ -17,16 +17,2853 @@
+         * \param path A string to append to the HKEY to attempt to open
+         *        can be nullptr if not needed
+         * \param[out] hkey A reference to a wil::unique_hkey to receive the opened HKEY
+         * \param access The requested access desired for the opened key
+         * \return HRESULT error code indicating success or failure (does not throw C++ exceptions)
+         */
+        inline HRESULT create_unique_key_nothrow(HKEY key, PCWSTR path, ::wil::unique_hkey& hkey, ::wil::reg::key_access access = ::wil::reg::key_access::read) WI_NOEXCEPT
+        {
+            hkey.reset();
+            const reg_view_details::reg_view_nothrow regview{ key };
+            return regview.create_key(path, hkey.addressof(), access);
+        }
+
+#if defined(__WIL_WINREG_STL)
+        /**
+         * \brief Opens a new HKEY to the specified path - see RegOpenKeyExW
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param path A string to append to the HKEY to attempt to open
+         *        can be nullptr if not needed
+         * \param[out] hkey A reference to a wil::shared_hkey to receive the opened HKEY
+         * \param access The requested access desired for the opened key
+         * \return HRESULT error code indicating success or failure (does not throw C++ exceptions)
+         */
+        inline HRESULT open_shared_key_nothrow(HKEY key, _In_opt_ PCWSTR path, ::wil::shared_hkey& hkey, ::wil::reg::key_access access = ::wil::reg::key_access::read) WI_NOEXCEPT
+        {
+            hkey.reset();
+            const reg_view_details::reg_view_nothrow regview{ key };
+            return regview.open_key(path, hkey.addressof(), access);
+        }
+
+        /**
+         * \brief Creates a new HKEY to the specified path - see RegCreateKeyExW
+         * \param key An opened registry key, or fixed registry key as the base key, from which to append the path
+         * \param path A string to append to the HKEY to attempt to open
+         *        can be nullptr if not needed
+         * \param[out] hkey A reference to a wil::shared_hkey to receive the opened HKEY
+         * \param access The requested access desired for the opened key
+         * \return HRESULT error code indicating success or failure (does not throw C++ exceptions)
+         */
+        inline HRESULT create_shared_key_nothrow(HKEY key, _In_opt_ PCWSTR path, ::wil::shared_hkey& hkey, ::wil::reg::key_access access = ::wil::reg::key_access::read) WI_NOEXCEPT
+        {
+            hkey.reset();
+            const reg_view_details::reg_view_nothrow regview{ key };
+            return regview.create_key(path, hkey.addressof(), access);
+        }
+#endif // #define __WIL_WINREG_STL
+
+        namespace reg_query_info_key_args
+        {
+            // RegQueryInfoKeyW has a long list of optional parameters
+            static constexpr LPWSTR null_class{ nullptr };

```

Comment:
LP-prefix is Hungarian Notation which is very passe' and the Coding Guidelines say DON'T

Global search/replace...
* `LPWSTR` -> `PWSTR`
* `LPCWSTR` -> `PCWSTR`
* `LPDWORD` -> `DWORD*`
* `PFILETIME` -> `FILETIME*`

and if there's any others too.

Only seemingly-HungarianNotation you MUST use nowadays is `PSID` as the actual definition may seem to be `void*` but it's actually not (for complex software archaeology reasons).

Reply:
Yep. I think I fixed them all.